server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -187,14 +187,15 @@ public void parse(DocumentParserContext context) throws IOException {
                 } else if (token == Token.VALUE_NUMBER || token == Token.VALUE_STRING) {
                     final String key = name() + "." + feature;
                     float value = context.parser().floatValue(true);
-                    if (context.doc().getByKey(key) != null) {
-                        throw new IllegalArgumentException(
-                            "[sparse_vector] fields do not support indexing multiple values for the same feature ["
-                                + key
-                                + "] in the same document"
+
+                    // if we have an existing feature of the same name we'll select for the one with the max value
+                    // based on recommendations from this paper: https://arxiv.org/pdf/2305.18494.pdf
+                    context.doc()
+                        .addOrUpdateWithKey(
+                            key,
+                            new FeatureField(name(), feature, value),
+                            (a, b) -> ((FeatureField) a).getFeatureValue() > ((FeatureField) b).getFeatureValue() ? a : b
                         );
```

Comment:
I am not sure we want to create a new field on every multi-value. 

Could we adjust the function definition to only be `calculateWithKey`, and it will pass `null` if it doesn't already exist. At that point a new one will be created and used, otherwise, we can call `setFeatureValue` on the existing one.

Reply:
Take a look at what I've got now (can always roll back).  But utilizing `setFeatureValue` (which I didn't realize was a good option here) let me drastically simplify the code.  LuceneDocument no longer has changes, no removal of features at all, and the logic can all sit sin SparseVectorFieldMapper.  

I may have missed some nuance of your suggestion here or some obvious requirement.  But as long as `setFeatureValue` has no detrimental downstream affects then it seems like a good solution?  

The concerns I had brought up in separate comment are related to how getting fields will return the first field by name but all of the sparse vector fields will have the same name.  Not sure if this is worth addressing here (vs creating a ticket) vs irrelevant but didn't want to lose that conversation.  We can discuss separately.